pkg/app/carbonapi/http_handlers.go

```diff
@@ -939,39 +940,148 @@ func (app *App) findHandler(w http.ResponseWriter, r *http.Request, lg *zap.Logg
 		logLevel = zapcore.ErrorLevel
 		return
 	}
-
+	// TODO: extend writeResponse w/new formats and migrate to it
 	if contentType == jsonFormat && jsonp != "" {
-		w.Header().Set("Content-Type", contentTypeJavaScript)
-		if _, writeErr := w.Write([]byte(jsonp)); writeErr != nil {
-			toLog.HttpCode = 499
-			logLevel = zapcore.WarnLevel
-			return
-		}
-		if _, writeErr := w.Write([]byte{'('}); writeErr != nil {
+		if writeResponse(ctx, w, blob, jsonFormat, jsonp) != nil {
 			toLog.HttpCode = 499
 			logLevel = zapcore.WarnLevel
 			return
 		}
+	} else {
+		w.Header().Set("Content-Type", contentType)
 		if _, writeErr := w.Write(blob); writeErr != nil {
 			toLog.HttpCode = 499
 			logLevel = zapcore.WarnLevel
 			return
 		}
-		if _, writeErr := w.Write([]byte{')'}); writeErr != nil {
-			toLog.HttpCode = 499
-			logLevel = zapcore.WarnLevel
-			return
+	}
+
+	toLog.HttpCode = http.StatusOK
+}
+
+func expandEncoder(globs dataTypes.Matches, leavesOnly string, groupByExpr string) ([]byte, error) {
```

Comment:
I suggest changing `leavesOnly` to a boolean. the boolean value could be assigned to the boolean variable at the start of expandHandler.

Reply:
Same for `groupByExpr`

Reply:
Why, if it's defined in API as a string? Not sure if I understand your suggestion clearly, though...

Reply:
Ah, disregard that. It's defined as integer, probably will change it to int then.

Reply:
No, you're right bool is better. Please check.